### PR TITLE
fix(@nestjs/event-emitter): allow non array event metadata

### LINKED ISSUE #751: Custom event decorators break with the latest array metadata type requirement
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

When using a custom event decorator an error is thrown: `eventListenerMetadatas is not iterable`. After digging it is clear that the latest release `1.4.1` and specifically this PR https://github.com/nestjs/event-emitter/pull/657 causes it.

### Minimum reproduction code

https://github.com/davidzwa/nest-event-bug/actions/runs/4292223946/jobs/7478397448

### Steps to reproduce

1) Create a custom decorator https://github.com/davidzwa/nest-event-bug/blob/main/src/custom-event.decorator.ts
2) Register it https://github.com/davidzwa/nest-event-bug/blob/9d616087d86bac5adbe0fac929eb373e180572bf/src/app.service.ts#L6

See MR repo workflow for the error https://github.com/davidzwa/nest-event-bug/actions/runs/4292223946/jobs/7478397448

### Expected behavior

The array type should either be enforced with the return type of the decorator (now using generic `MethodDecorator`) or **preferably** it should rather be backwards compatible and allow non-array event metadata to work without failure.

Undesirable alternative:
The error `eventListenerMetadatas is not iterable` should be prevented and clarified.

Takeaway: the return type at https://github.com/nestjs/event-emitter/blob/f9667ae391eb47c17ab855d10ae6514ceb1876a6/lib/events-metadata.accessor.ts#L13 is not enforced at build time.

### Package version

1.4.1

### NestJS version

^9.0.0

### Node.js version

18.14.2

### In which operating systems have you tested?

- [ ] macOS
- [X] Windows
- [X] Linux

### Other

_No response_

### PR DESCRIPTION
**Pull request by Maligosus**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: Resolves #751 

## What is the new behavior?
Allow to use non array metadata.

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by Maligosus** [comment]
This PR also included changes from https://github.com/nestjs/event-emitter/pull/749

File: lib/event-subscribers.loader.ts [hunk 1]
```diff
@@ -50,17 +50,16 @@ export class EventSubscribersLoader
         const { instance } = wrapper;
         const prototype = Object.getPrototypeOf(instance) || {};
         const isRequestScoped = !wrapper.isDependencyTreeStatic();
-        this.metadataScanner.scanFromPrototype(
-          instance,
-          prototype,
-          (methodKey: string) =>
+        this.metadataScanner
```
Comment (kamilmysliwiec) [inline lib/event-subscribers.loader.ts:53]: Can you revert this change? This is a breaking one for those that are not on the latest version of @nestjs/core

Comment (Maligosus) [reply lib/event-subscribers.loader.ts]: > This PR also included changes from #749
revert

Comment (Maligosus) [reply lib/event-subscribers.loader.ts]: Ready.

File: lib/events-metadata.accessor.ts [hunk 2]
```diff
@@ -10,6 +10,8 @@ export class EventsMetadataAccessor {
   getEventHandlerMetadata(
     target: Type<unknown>,
   ): OnEventMetadata[] | undefined {
-    return this.reflector.get(EVENT_LISTENER_METADATA, target);
+    const metadata = this.reflector.get(EVENT_LISTENER_METADATA, target);
+    if (!metadata) return undefined;
```
Comment (kamilmysliwiec) [inline lib/events-metadata.accessor.ts:14]:
```suggestion
    if (!metadata) {
       return undefined;
    }
```
